libdnf/module/modulemd/ModulePackage.cpp

```diff
@@ -309,13 +316,39 @@ std::string ModulePackage::getDescription() const
  */
 std::vector<std::string> ModulePackage::getArtifacts() const
 {
-    return metadata.getArtifacts();
+    std::vector<std::string> result_rpms;
+    char ** rpms = modulemd_module_stream_v2_get_rpm_artifacts_as_strv((ModulemdModuleStreamV2 *) mdStream);
+
+    for (char **iter = rpms; iter && *iter; iter++) {
+        result_rpms.emplace_back(std::string(*iter));
+        g_free(*iter);
+    }
+
+    g_free(rpms);
+    return result_rpms;
 }
 
 std::vector<ModuleProfile>
 ModulePackage::getProfiles(const std::string &name) const
 {
-    return metadata.getProfiles(name);
+    std::vector<ModuleProfile> result_profiles;
+    char ** profiles = modulemd_module_stream_v2_get_profile_names_as_strv((ModulemdModuleStreamV2 *) mdStream);
```

Comment:
This looks like the sort of place that we probably need a new function in libmodulemd. Previously, I was only asked about looking up NEVRA by glob, but if you need to support finding profiles by glob as well, I could add a `ModuleStreamV2.search_profiles(const gchar *profile_pattern)` function. Should I do that? There's still time to get it into the 2.9.0 release.

Reply:
It would be nice. :slightly_smiling_face: